src/Events/Telemetry/Provider.php

```diff
@@ -0,0 +1,79 @@
+<?php
+/**
+ * Service Provider for interfacing with TEC\Common\Telemetry.
+ *
+ * @since   TBD
+ *
+ * @package TEC\Events\Telemetry
+ */
+
+namespace TEC\Events\Telemetry;
+
+use TEC\Common\Contracts\Service_Provider;
+
+ /**
+  * Class Provider
+  *
+  * @since   TBD
+
+  * @package TEC\Events\Telemetry
+  */
+class Provider extends Service_Provider {
+	public function register() {
+		$this->add_filters();
+	}
+
+	public function add_filters() {
+		add_filter( 'tec_common_telemetry_optin_args', [ $this, 'filter_tec_common_telemetry_optin_args' ] );
+		add_filter( 'tribe_general_settings_debugging_section',[ $this, 'filter_tribe_general_settings_debugging_section' ] );
+		add_filter( 'tribe_field_value', [ $this, 'filter_tribe_field_opt_in_status' ], 10, 2 );
+		add_filter( 'tec_telemetry_slugs', [ $this, 'filter_tec_telemetry_slugs' ] );
+	}
+
+	public function filter_tec_common_telemetry_optin_args( $optin_args ) {
```

Comment:
```suggestion
	/**
	 * Filter the telemetry optin arguments.
	 *
	 * @since TBD
	 *
	 * @param array $optin_args Previous set of args we are changing.
	 *
	 * @return array 
	 */
	public function filter_tec_common_telemetry_optin_args( $optin_args ) {
```